hbase_sql/src/main/java/com/splicemachine/derby/stream/spark/SparkExportDataSetWriter.java

```diff
@@ -93,16 +98,23 @@ public void readExternal(ObjectInput in) throws IOException, ClassNotFoundExcept
         job.setOutputFormatClass(SparkDataSet.EOutputFormat.class);
         job.getConfiguration().set("mapred.output.dir",directory);
 
-        JavaRDD<V> cached=rdd.cache();
-        long writtenRows=cached.count();
+        SparkContext context = rdd.context();
+        AtomicLong writtenRows = new AtomicLong(0);
+        context.addSparkListener(new SparkListener() {
+            @Override
+            public void onTaskEnd(SparkListenerTaskEnd taskEnd) {
+                writtenRows.getAndAdd(taskEnd.taskMetrics().outputMetrics().recordsWritten());
+            }
+        });
+
         rdd.keyBy(new NullFunction<V>())
             .setName(String.format("Export Directory: %s", directory))
             .saveAsNewAPIHadoopDataset(job.getConfiguration());
-        cached.unpersist();
 
+        long end = System.currentTimeMillis();
         ValueRow valueRow=new ValueRow(2);
-        valueRow.setColumn(1,new SQLLongint(writtenRows));
-        valueRow.setColumn(2,new SQLInteger(0));
+        valueRow.setColumn(1,new SQLLongint(writtenRows.get()));
+        valueRow.setColumn(2,new SQLLongint(end-start));
```

Comment:
If we change the second column to be the elapsed time here, do we need to change ControlExportDataSetWriter to make them consistent?